backend/danswer/danswerbot/slack/handlers/handle_standard_answers.py

```diff
@@ -150,11 +150,13 @@ def handle_standard_answers(
 
         formatted_answers = []
         for standard_answer in matching_standard_answers:
-            block_quotified_answer = ">" + standard_answer.answer.replace("\n", "\n> ")
-            formatted_answer = (
-                f'Since you mentioned _"{standard_answer.keyword}"_, '
-                f"I thought this might be useful: \n\n{block_quotified_answer}"
+            since_you_mentioned_pretext = (
+                f'Since you mentioned _"{standard_answer.keyword}"_'
+                if not standard_answer.match_regex
+                else f"Since your question matches the regex pattern `{standard_answer.keyword}`"
```

Comment:
Putting my end user hat on, I feel like we shouldn't be showing regex's to end users ... was this was requested?  Seems much too technical.

Reply:
Gonna defer to @yuhongsun96 on this.

Reply:
Was not requested, I think this probably will not add much value to the user but I don't hate it either. I think something more generic could work. We could grab the match group from the regex and show that instead. Like since your question contains "wifi password"... (in the regex case)

Reply:
Showing the matched result sounds more reasonable.

Reply:
Alright I'll make that change.

Reply:
After talking with @yuhongsun96 I think it would be good to keep the matching logic in one place. Currently,  `find_matching_standard_answers` performs the matching, so I think extracting the matched string there makes sense.

In the interest of avoiding scope creep, I think it may make sense as a stopgap measure to reintroduce the `tuple[StandardAnswer, str]` response type, and then perform a more thorough refactor with a Pydantic type as a separate PR.

Reply:
I threw together a "fast but messy" implementation here to resolve this comment. We can decide how to prioritize the refactor separately.